### ISSUE TITLE
Auto full screen needs restart

### ISSUE THREAD

**Issue by dsl101** [Knowinnovation/bluetime#32]
When changing the option to auto full screen, closing and reopening the app doesn't seem to be enough. I had to force close and relaunch to get it to open full screen.

**Comment by doctor3w** [comment]
Yes, the auto full screen only opens to full screen on a new app launch. Not upon closing and reopening the app because I figured one could just do that manually at the point if you're already in the app.

**Comment by dsl101** [comment]
True - just perhaps confusing to set it, go full screen manually, close the
app, then relaunch later and it _not_ be full screen as you have set in
prefs.

**Comment by doctor3w** [comment]
Well if one exits in full screen it should reopen in fullscreen. The only time that Auto Open in fullscreen applies is after termination, then it opens in full screen. I thought it would be annoying for a user if, say he has the setting on, exits full screen to change a setting or the time or something, closes the app while doing that for whatever reason, and then open it up to find he's back in full screen and not at the same place he left the app.

**Comment by dsl101** [comment]
That also makes sense. So, what is the use case for that option - rather
than as you said just open in last used mode?

**Comment by doctor3w** [comment]
So iPhones apps have two close states, backgrounded and terminated. A backgrounded app lives for 10 minutes in the background frozen in RAM. Terminated removes it completely from RAM. The use case of this function is for the terminated stage. So if I completely kill the application, upon next open it'll be in full screen. If I just background the application, it'll open to the last screen. If you background while in full screen, it should remain in full screen upon resuming active. 
The reason for this usage is because like I said before if someone has exited full screen, left the app for a moment and then resumed, they probably want to continue where they left off. Also, assuming you have a display device (an iPad say), the most likely usage is to open the app, it automatically enters full screen with the setting on, you place it down, connect your phone and leave it there. Not touching or editing the time from it. 

---

Drew Dunne
drew@drewsdunne.com
http://www.drewsdunne.com

On Tue, Jul 5, 2016 at 1:07 PM -0400, "dsl101" notifications@github.com wrote:

That also makes sense. So, what is the use case for that option - rather

than as you said just open in last used mode?

**Comment by dsl101** [comment]
I'm still not sure where 'resume last state' doesn't cover those cases
though - which is fine. In the iPad example, if you always use it as the
display device, it will always be full screen when you close the app, so
would always open up like that next time using 'last state', no? I haven't
come across any apps which distinguish cold start from background start
before, which is probably why I was confused.

**Comment by doctor3w** [comment]
## Most apps, if you kill it by swiping up on the multitasking card, will then reopen to their main screen. For example I use outlook. If I am reading an email, kill the app, and reopen it, it opens to my email inbox, not the email I was reading. 

Drew Dunne
drew@drewsdunne.com
http://www.drewsdunne.com

On Wed, Jul 6, 2016 at 2:45 AM -0400, "dsl101" notifications@github.com wrote:

I'm still not sure where 'resume last state' doesn't cover those cases

though - which is fine. In the iPad example, if you always use it as the

display device, it will always be full screen when you close the app, so

would always open up like that next time using 'last state', no? I haven't

come across any apps which distinguish cold start from background start

before, which is probably why I was confused.

**Comment by andyburnett** [comment]
I think I agree with Drew. When I kill a app, I think what I'm trying to do is reset it. Normally, this is because it crashed in a horrible heap. But, irrespective of the reason, I think I'm expecting it to go back to its default state.

Of course, I don't expect it to lose all its settings, so the definition of what we mean by default state is another matter entirely. For instance, I wouldn't expect it's fonts to have reverted to whatever the out-of-the-box experience was.

**Comment by doctor3w** [comment]
So in this case, that settings sets the "default" state to full screen. 

---

Drew Dunne
drew@drewsdunne.com
http://www.drewsdunne.com

On Wed, Jul 6, 2016 at 9:54 AM -0400, "andyburnett" notifications@github.com wrote:

I think I agree with Drew. When I kill a app, I think what I'm trying to do is reset it. Normally, this is because it crashed in a horrible heap. But, irrespective of the reason, I think I'm expecting it to go back to its default state.

Of course, I don't expect it to lose all its settings, so the definition of what we mean by default state is another matter entirely. For instance, I wouldn't expect it's fonts to have reverted to whatever the out-of-the-box experience was.

**Comment by dsl101** [comment]
I think what I meant about it being 'different' to other apps is that in
this case, the setting _only_ affects cold start not warm start, and I
don't know many other apps which distinguish those two cases in their
settings. The example being that I set 'Auto full screen', closed the app
and opened it again expecting it to go full screen. When it didn't, I
thought it was broken. So perhaps it's just a labelling thing?

**Comment by doctor3w** [comment]
I guess that's just a difference in how it one defines closing the app.
